packages/opensrp-store/src/store/selectors.ts

```diff
@@ -0,0 +1,38 @@
+import { reducerName as sessionReducerName } from '@onaio/session-reducer';
+import { Store } from 'redux';
+import { Dictionary } from '@onaio/utils';
+
+/** get API Token from the Redux store
+ * @param {Partial<Store>} state - the redux store
+ */
+interface StoreState {
+  session: {
+    [key: string]: Dictionary;
+  };
+}
+export function getApiToken(state: Partial<Store>): string {
+  const { extraData } = (state as StoreState)[sessionReducerName];
+  return extraData.api_token || null;
+}
+
+/** get Access Token from the Redux store
+ * @param {Partial<Store>} state - the redux store
+ */
+export function getAccessToken(state: Partial<Store>): string | null {
+  const { extraData } = (state as StoreState)[sessionReducerName];
+  if (extraData.oAuth2Data && extraData.oAuth2Data.access_token) {
+    return extraData.oAuth2Data.access_token;
+  }
+  return null;
+}
+
+/** get the oAuth2 provider state parameter from the Redux store
+ * @param {Partial<Store>} state - the redux store
+ */
+export function getOauthProviderState(state: Partial<Store>): string | null {
+  const { extraData } = (state as StoreState)[sessionReducerName];
+  if (extraData.oAuth2Data && extraData.oAuth2Data.state) {
+    return extraData.oAuth2Data.state;
+  }
+  return null;
+}
```

Comment:
We need to use reselect

Reply:
Going to track this here https://github.com/OpenSRP/web/issues/63